File: src/main/java/com/lls/app/hdfs/HdFsApplication.java
package com.lls.app.hdfs;

import org.apache.spark.SparkConf;
import org.apache.spark.SparkContext;
import org.apache.spark.rdd.RDD;
import scala.Function1;
import scala.runtime.BoxedUnit;

/************************************
 * HdFsApplication
 * @author liliangshan
 * @date 2020/7/25
 ************************************/
public class HdFsApplication {

    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
                .setMaster("spark://liliangshan.local:7077")
                .setAppName("HdFsApplication");
        SparkContext context = new SparkContext(conf);
        RDD<String> rdd = context.textFile("hdfs://liliangshan.local:9000/input/ali-test.kubeconfig", 2);
    }

}

File: src/main/scala/com/lls/sc/hdfs/HdFsApplication.scala
package com.lls.sc.hdfs

import org.apache.spark.{SparkConf, SparkContext}

object HdFsApplication {

//    读取hdfs文件数据
    def main(args: Array[String]): Unit = {
        val conf = new SparkConf().setMaster("local[*]").setAppName("HdFsApplication")
        val sparkContext = new SparkContext(conf)
        val lines = sparkContext.textFile("hdfs://liliangshan.local:9000/input/ali-test.kubeconfig")
        lines.foreach(println(_))
        sparkContext.stop()
    }

}
